clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoAdminClient.java

```diff
@@ -157,6 +180,264 @@ public boolean dropMetalake(NameIdentifier ident) {
+   * @throws RuntimeException If adding the Group encounters storage issues.
+   */
+  public Group addGroup(String metalake, String group) throws GroupAlreadyExistsException {
+    GroupAddRequest req = new GroupAddRequest(group);
+    req.validate();
+
+    GroupResponse resp =
+        restClient.post(
+            String.format(API_METALAKES_GROUPS_PATH, metalake, ""),
+            req,
+            GroupResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.groupErrorHandler());
+    resp.validate();
+
+    return resp.getGroup();
+  }
+
+  /**
+   * Removes a Group.
+   *
+   * @param metalake The Metalake of the Group.
+   * @param group THe name of the Group.
+   * @return `true` if the Group was successfully removed, `false` otherwise.
+   * @throws RuntimeException If removing the Group encounters storage issues.
+   */
+  public boolean removeGroup(String metalake, String group) {
+    RemoveResponse resp =
+        restClient.delete(
+            String.format(API_METALAKES_GROUPS_PATH, metalake, group),
+            RemoveResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.groupErrorHandler());
+    resp.validate();
+
+    return resp.removed();
+  }
+
+  /**
+   * Gets a Group.
+   *
+   * @param metalake The Metalake of the Group.
+   * @param group THe name of the Group.
+   * @return The getting Group instance.
+   * @throws NoSuchGroupException If the Group with the given identifier does not exist.
+   * @throws RuntimeException If getting the Group encounters storage issues.
+   */
+  public Group getGroup(String metalake, String group) throws NoSuchGroupException {
+    GroupResponse resp =
+        restClient.get(
+            String.format(API_METALAKES_GROUPS_PATH, metalake, group),
+            GroupResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.groupErrorHandler());
+    resp.validate();
+
+    return resp.getGroup();
+  }
+
+  /**
+   * Adds a new metalake admin.
+   *
+   * @param user The name of the User.
+   * @return The added User instance.
+   * @throws UserAlreadyExistsException If a User with the same identifier already exists.
+   * @throws RuntimeException If adding the User encounters storage issues.
+   */
+  public User addMetalakeAdmin(String user) {
+    UserAddRequest req = new UserAddRequest(user);
+    req.validate();
+
+    UserResponse resp =
+        restClient.post(
+            String.format(API_ADMIN_PATH, ""),
+            req,
+            UserResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.userErrorHandler());
+    resp.validate();
+
+    return resp.getUser();
+  }
+
+  /**
+   * Removes a metalake admin.
+   *
+   * @param user The name of the User.
+   * @return `true` if the User was successfully removed, `false` otherwise.
+   * @throws RuntimeException If removing the User encounters storage issues.
```

Comment:
I think the exception you mentioned here is not correct, it is not just `RuntimeException` that will be thrown.